service-base/src/main/java/org/eclipse/hono/service/HealthCheckServer.java

```diff
@@ -87,6 +107,60 @@ private void prepareHealthCheck() {
         }
     }
 
+    private void registerConsumers() {
+        readinessConsumer = vertx.eventBus().consumer(EVENT_BUS_ADDRESS_HEALTH_READINESS);
+        readinessConsumer.handler(this::processReadinessMessage);
+        LOG.info("listening on event bus [address: {}] for readiness status", EVENT_BUS_ADDRESS_HEALTH_READINESS);
+
+        livenessConsumer = vertx.eventBus().consumer(EVENT_BUS_ADDRESS_HEALTH_LIVENESS);
+        livenessConsumer.handler(this::processLivenessMessage);
+        LOG.info("listening on event bus [address: {}] for liveness status", EVENT_BUS_ADDRESS_HEALTH_LIVENESS);
+    }
+
+    private void processReadinessMessage(final Message<JsonObject> msg) {
+
+        final String verticleName = msg.headers().get(MSG_FIELD_VERTICLE_NAME);
+        final String instanceId = msg.headers().get(MSG_FIELD_VERTICLE_INSTANCE_ID);
+        final String procedureName = msg.headers().get(MSG_FIELD_PROCEDURE_NAME);
+        final Status status = new Status(msg.body());
+
+        LOG.info("received readiness update {}/{}/{}: {}", verticleName, instanceId, procedureName, status.isOk());
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("readiness update status: {}", msg.body().encodePrettily());
+        }
+
+        readinessHandler.register(verticleName + "/" + instanceId + "/" + procedureName,
```

Comment:
I am not sure if this method is supposed to be called *dynamically* during run time. My understanding is that it should be invoked during startup to set up the checks once. My concern is that this method is quite costly to invoke but I haven't checked.

Reply:
It is probably not supposed to be used that way. But on the other hand I saw no problem doing so. It contains a synchronized block which should be no problem since we have only one instance of the class using it. I expect it to be not more expensive than managing the data structures for all results ourselves, whereas the event bus usage should be much more costly.